jlab-sensing/soil-power-sensor-firmware
# Move user code to separate libraries

**Name/Affiliation/Title**
John Madden, UCSC, Maintainer

**Purpose of the PR**
Move all our "user" code to individual libraries so that it's easier for new contributors to understand the project structure. All the generated stm32cubemx code will live in `Src` and `Inc`.

**Development Environment**
`Linux spruce 6.8.2-arch2-1 #1 SMP PREEMPT_DYNAMIC Thu, 28 Mar 2024 17:06:35 +0000 x86_64 GNU/Linux`
PlatformIO Core, version 6.1.7
Used ST-Link V3 MINIE

**Test Procedure**
Check that all the envs and tests can build

```
pio run -t clean
pio run
pio test -e tests --without-uploading --without-testing
```

**Additional Context**
I think this PR will be very helpful for new colaborators as they do not need to sift through a huge base to find the information relevant to our project.

Closes #36
